### PR TITLE
Adds custom title and link values for related news and events listings

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Ticket [2454](https://projects.torchbox.com/projects/rca-django-cms-project/tickets/2454)

image

### CODE REVIEW

File: rca/landingpages/models.py [hunk 1]
```diff
@@ -291,13 +306,34 @@ class LandingPage(TapMixin, ContactFieldsMixin, LegacyNewsAndEventsMixin, BasePa
...
         ),
     ]
 
+    @property
+    def news_view_all(self):
+        return {
+            "link": self.news_and_events_link_target_url,
+            "title": self.news_and_events_link_text,
+        }
+
```
Comment (jams2) [inline rca/landingpages/models.py:336]: Suggestion - there's quite a bit of repetition with the field definitions, helper property and content panel definitions. Could we refactor this into a mixin class? (the content panels could be returned from a method that takes extra `FieldPanels` as arguments to get around the `legacy_news_and_event_tags` not being on all Page types)

File: rca/project_styleguide/templates/patterns/organisms/news/news_api_content.html [hunk 2]
```diff
@@ -16,8 +16,8 @@
         {% endif %}
 
         {% if forloop.last and not forloop.first %}
-                <a class="news__view-all link link--primary link--link body body--two" href="{{ news_view_all.link|default:'/news-and-events/' }}">
-                    <span class="link__label">{{ news_view_all.title|default:'See all news & events' }}</span>
+                <a class="news__view-all link link--primary link--link body body--two" href="{{ page.news_view_all.link|default:'/news-and-events/' }}">
```
Comment (jams2) [inline rca/project_styleguide/templates/patterns/organisms/news/news_api_content.html:19]: Question - are we confident that the default /news-and-events/ route will always resolve? A default page selected on a settings model seems more robust.

File: rca/home/models.py [hunk 3]
```diff
@@ -157,7 +158,19 @@ class HomePage(TapMixin, BasePage):
 
     use_api_for_alumni_stories = models.BooleanField(default=True)
     use_api_for_news_and_events = models.BooleanField(default=True)
-
+    news_and_events_link_text = models.TextField(
+        max_length=120,
+        blank=True,
+        help_text=_("The text do display for the 'View all news and events' link"),
+    )
+    news_and_events_link_target_url = models.URLField(
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/home/models.py:166]: We do have a habit of doing this for RCA it's so they can add links to pre-filtered listing pages, E.G

'https://www.rca.ac.uk/news-and-events/news/?programme=architecture-pathway-mres-rca#results'

File: rca/project_styleguide/templates/patterns/organisms/news/news_api_content.html (hunk 2 above)
Comment (jams2) [reply rca/project_styleguide/templates/patterns/organisms/news/news_api_content.html:19]: Given the project convention for using URLFields I'd say this is fine then

**Pull request merged**